### ISSUE TITLE
[ENH]: Updates for Display Class in PyCaret 3.0

### ISSUE THREAD

**Issue by moezali1** [pycaret/pycaret#2479]
### Describe the feature you want to add to this project

Display Class in PyCaret 2.x

# html parameter in the setup
There are two parameters in the setup `silent`, and `html` which are mostly used when pycaret is not used in the Notebook. `silent` parameter avoids the confirmation box after `setup` is executed. `silent` parameter has been removed from 3.0 already. For `html` parameter, the default should be set to 'auto' with automatic inference behind the scene. If `IPython` is not detected then the display class should adopt behavior that works well with the command line. Most functions have display bar which uses `IPython` and teh `compare_models` function has a `update_display` function that updates the table in real time, this is also an `IPython` functionality. 

Expected Fix:
- Default value of `html` parameter set to 'auto' by default. User can still over-write it explicitly with `True` or `False` boolean.
- When IPython is not available the output of functions shouldn't print IPython progress bars, instead it should print tqdm progress bar which is fully compatible with command line and IDE's like Spyder.
- When IPython is not available `compare_models` output should be printed once the run is finished. At the moment, it prints 1 model per print with headers, so if there are 15 models in the `compare_models` run, you will see 15 print statements.
- When IPython is available, it should proceed to do what it is currently doing with `html=True`. 

# Databricks
The output in Databricks is broken at the moment because of lack of IPython availability. However Databricks can have formated pandas print but no IPython activity. 

Expected Fix: We should automatically infer if user is using databricks (if we can't automatically infer, we can also take that as an input parameter in the setup. (PS - We can think about renaming `html` parameter to something more general. 3.0-rc is the good time to make that change if we have to).

When user is using Databricks:
- get_data should print formatted pandas dispaly
- setup should print formatted pandas table. I think styler is not possible in Databricks yet (I am okay with without color output as well).
- create_model, tune_model, etc. should print formatted pandas output after the function fnishes
- compare_models should print the output in formatted pandas dataframe (if styler not possible) after function finishes
- Instead of progress bar everywhere, we should use tqdm which is likely Databricks compatible (similar to what we are doing for IDE's like spyder).

# Supporting Images

![image](https://user-images.githubusercontent.com/54699234/166112004-619cbf21-7b54-4df1-a884-42bf8bdda8c8.png)
![image](https://user-images.githubusercontent.com/54699234/166112000-d71875fa-3bf0-404c-bda9-3a057dcd498b.png)
![image](https://user-images.githubusercontent.com/54699234/166111997-c694c1aa-f278-45b4-ba92-621cdf1f1ad5.png)
![image](https://user-images.githubusercontent.com/54699234/166111990-1fbb4287-0e5a-4cd2-af50-c6333ccc1f0f.png)
![image](https://user-images.githubusercontent.com/54699234/166111986-e2b81a46-b397-4db5-a74b-f72e2938a7ca.png)

### Describe your proposed solution

Explained above

**Comment by moezali1** [comment]
@Yard1 I tested `git+https://github.com/pycaret/pycaret.git@display_enhancements` on Databricks and here are my observations:

## get_data function

The output is still unformatted. See below:
![image](https://user-images.githubusercontent.com/54699234/169189309-36709fb1-f1fc-4f23-b6b0-829e9118cb91.png)
![image](https://user-images.githubusercontent.com/54699234/169189335-52f6c081-7590-467a-ac3c-22f088459864.png)

## setup function
The output is unformatted. See below:

![image](https://user-images.githubusercontent.com/54699234/169189383-0b64b73b-70b4-462b-a6a5-fdcdc44bbc32.png)
![image](https://user-images.githubusercontent.com/54699234/169189392-5072bc7c-219a-45c8-9aee-f0d9552f7efa.png)

## create_model function
The output is unformatted. See below:
![image](https://user-images.githubusercontent.com/54699234/169189431-fcae12bf-0ab4-427a-be7f-eaea0e917295.png)
![image](https://user-images.githubusercontent.com/54699234/169189442-749ce629-369c-4316-89c4-82c3f3793220.png)

## tune_model, ensemble_model, other functions
Same issue as above

## compare_models
For every model update, it prints the header of the table again and again. 
![image](https://user-images.githubusercontent.com/54699234/169189499-7078e8f3-b32b-4740-934f-6449a539ed4a.png)

**Comment by moezali1** [comment]
Another issue is when I test this on colab it prints tqdm progress bar instead of Ipython. This is not expected behavior. 

![image](https://user-images.githubusercontent.com/54699234/169189752-6a8999ab-5103-4654-a79c-13486a8d96da.png)

![image](https://user-images.githubusercontent.com/54699234/169189759-6d2ca902-0b8a-4efd-9de7-274cf100b8b9.png)

**Comment by Yard1** [comment]
Hey, quick question, did you detach and reattach after installing?

**Comment by Yard1** [comment]
Colab now uses tqdm progress bar as well, but it uses it in IPython format.

**Comment by moezali1** [comment]
@Yard1 Yes I detached and reattached. 

The problem with tqdm bar on colab is they won't disappear like IPython :(

**Comment by Yard1** [comment]
I can make it disappear!

**Comment by moezali1** [comment]
Is there an option to keep progress bar same as what we show in Jupyter. It will be good to have colab consistent with jupyter. What do you think?

**Comment by moezali1** [comment]
@tvdboom @ngupta23 please feel free to add your views.

**Comment by moezali1** [comment]
One issue remaining: 

On colab the progress bar is different color and won't go away after plot is displayed. All other functions its behaving correctly.

![image](https://user-images.githubusercontent.com/54699234/169700241-c8d2986f-bdec-46aa-b7e0-771376be2ba4.png)

![image](https://user-images.githubusercontent.com/54699234/169700250-e7ed9c54-9dd8-41b3-877f-cd94275d8a1e.png)

**Comment by moezali1** [comment]
When I run `compare_models` on the command line or execute python files through VS code terminal. I see a lot of warnings. I can hardly see the results of `compare_models` until it finishes. I tried setting `warnings.simplefilter("ignore")` in my code but it didn't help.

Expected Fix: We should be able to silent these warnings. 

![image](https://user-images.githubusercontent.com/54699234/169704903-876b0930-e40d-471c-94e8-56d5823cc6c8.png)

**Comment by Yard1** [comment]
@moezali1 Updated, PTAL and retest
